sound/soc/soc-topology.c

```diff
@@ -1364,6 +1364,10 @@ static struct snd_kcontrol_new *soc_tplg_dapm_widget_denum_create(
 
 	for (i = 0; i < num_kcontrols; i++) {
 		ec = (struct snd_soc_tplg_enum_control *)tplg->pos;
+
+		tplg->pos += (sizeof(struct snd_soc_tplg_enum_control) +
+				ec->priv.size);
+
```

Comment:
is this the right position?
In soc_tplg_dbytes_create, the pointer is modified after strlen+ kzalloc is checked. same in soc_tplg_dmixer_create.
But not in soc_tplg_denum_create. It's done between kzalloc and strlen check in soc_tplg_dapm_widget_dmixer_create(). 

it'd be good to have a more regular pattern across objects/widgets.

Reply:
@plbossart ok let me make it consistent across all controls

Reply:
@plbossart fixed now

Reply:
I don't see how this is aligned now, sorry, the location of the increment is different?